src/frontends/tensorflow/src/graph_iterator_proto.hpp

```diff
@@ -18,21 +18,59 @@ namespace frontend {
 namespace tensorflow {
 
 class GraphIteratorProto : public GraphIterator {
-    std::vector<const ::tensorflow::NodeDef*> m_nodes;
-    size_t node_index = 0;
     std::shared_ptr<::tensorflow::GraphDef> m_graph_def;
+    std::shared_ptr<::tensorflow::FunctionDef> m_func_def;
+
+    size_t node_index = 0;
+    std::vector<std::shared_ptr<DecoderBase>> m_decoders;
 
 public:
+    GraphIteratorProto(const std::shared_ptr<::tensorflow::GraphDef>& graph_def,
```

Comment:
do we plan to use this ctor when func_def is empty?
if so, we have to add additional checks

Reply:
Now this constructor will be used only for creation `GraphIterator` for `FunctionDef`. No plan to use it for standalone `GraphDef`.